tools/linter.py

```diff
@@ -35,7 +35,22 @@ def run_lint(self, fix: bool) -> None:
 
         errors and print(errors)
 
-        sys.exit(retcode)
+        # Running borrowed ref checker
```

Comment:
This now looks oddly asymmetric between calling the `run_ruff` method for one linter check and then an inline invocation for the other check (also with asymmetric stdout/stderr handling). I suggest cleaning this up by using a `run_check_c_api` method and also making that return `return res.returncode, res.stdout`.

Reply:
Thanks. Almost what I meant but not quite. I think it will look better if you'd leave the `if __name__ == "__main__":` block unchanged (i.e., just calling `DiffLinter().run_lint(fix=False)`) and the `run_lint` method would call `run_ruff` and `run_check_c_api`.

I suggest still tweaking that. Other than that, I'll give it a 👍🏼 and will leave the final merge to @ngoldbaum.